project-koku/koku
# add exception handling for malformed headers

While reviewing #1481, I was inadvertently creating malformed headers and the API was not handling exceptions. This PR handles malformed headers in middleware and raises a 403 Forbidden.

Testing:
1. set `DEVELOPMENT=False` in `.env`.
2. in Postman, create a new GET request to `api/cost-management/v1/providers/`. Under `Headers`, set a new (key: value) pair: 'x-rh-identity': eyJpZGVudGl0eSI6IHsiYWNjb3VudF9udW1iZXIiOiAiMTAwMDEiLCAidHlwZSI6ICJVc2VyIiwgInVzZXIiOiB7InVzZXJuYW1lIjogInVzZXJfZGV2IiwgImVtYWlsIjogInVzZXJfZGV2QGZvby5jb20iLCAiaXNfb3JnX2FkbWluIjogdHJ1ZX19LCAiZW50aXRsZW1lbnRzIjogeyJjb3N0X21hbmFnZW1lbnQiOiB7ImlzX2VudGl0bGVkIjogdHJ1ZX19fQ=='. Send request and confirm 200 response.
3. Create a new GET request to `api/cost-management/v1/providers/`.  Under `Headers` add the key: `x-rh-identity` and value: `malformed`. Send the request.
4. See 403 Forbidden.
